src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,415 @@
+                            {
+                                position++;
+                            }
+                            else
+                            {
+                                position--;
+                                break;
+                            }
+                        }
+
+                        if (fmt == null)
+                        {
+                            fmt = new StringBuilder();
+                        }
+                        fmt.Append(ch);
+                    }
+                }
+                if (ch != '}')
+                {
+                    FormatError();
+                }
+
+                string sFmt = null;
+                string s = null;
+
+                if (s == null)
+                {
+                    var formattableArg = arg as IFormattable;
+
+                    if (formattableArg != null)
+                    {
+                        if (sFmt == null && fmt != null)
+                        {
+                            sFmt = fmt.ToString();
+                        }
+
+                        s = formattableArg.ToString(sFmt, null);
+                    }
+                    else if (arg != null)
+                    {
+                        s = arg.ToString();
+                    }
+                }
+
+                if (s == null)
+                {
+                    s = string.Empty;
+                }
+                int pad = width - s.Length;
+                if (!leftJustify && pad > 0)
+                {
+                    builder.Append(' ', pad);
+                }
+
+                builder.Append(s);
+
+                if (leftJustify && pad > 0)
+                {
+                    builder.Append(' ', pad);
+                }
+
+                while (true)
+                {
+                    if (position == length)
+                    {
+                        FormatError();
+                    }
+                    ch = resourceString[position];
+                    position++;
+                    if (ch == '}')
+                    {
+                        if (position < length && resourceString[position] == '}')  // Treat as escape character for }}
+                        {
+                            builder.Append(ch);
+                            position++;
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                resourceStringBuilder.Append(EncodeArgument(builder.ToString()));
+            }
+
+            return resourceStringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Encodes the argument based on the object type.
+        /// </summary>
+        /// <param name="argument">The object to encode.</param>
+        /// <returns>The encoded object.</returns>
+        protected object EncodeArgument(object argument)
```

Comment:
Needs a new name since it's only called with a `string` argument now. In addition the special case for `HtmlString` is no longer relevant (though perhaps it should be?).